drivers/i2s/i2s_mcux_flexcomm.c

```diff
@@ -0,0 +1,897 @@
+		return -EIO;
+	}
+
+	ret = k_msgq_get(&strm->out_queue, &buffer,
+			 SYS_TIMEOUT_MS(strm->cfg.timeout));
+
+	if (ret != 0) {
+		if (strm->state == I2S_STATE_ERROR) {
+			return -EIO;
+		} else {
+			return -EAGAIN;
+		}
+	}
+
+	*mem_block = buffer;
+	*size = strm->cfg.block_size;
+	return 0;
+}
+
+static int i2s_mcux_write(const struct device *dev, void *mem_block,
+			   size_t size)
+{
+	struct i2s_mcux_data *dev_data = dev->data;
+	struct stream *strm = &dev_data->dma_tx;
+	int ret;
+
+
+	if (strm->state != I2S_STATE_RUNNING &&
+		strm->state != I2S_STATE_READY) {
+		LOG_ERR("invalid state (%d)", strm->state);
+		return -EIO;
+	}
+
+	ret = k_msgq_put(&strm->in_queue, &mem_block,
+			 SYS_TIMEOUT_MS(strm->cfg.timeout));
+
+	if (ret) {
+		LOG_ERR("k_msgq_put failed %d", ret);
+		return ret;
+	}
+
+	return ret;
+}
+
+static const struct i2s_driver_api i2s_mcux_driver_api = {
+	.configure = i2s_mcux_configure,
+	.config_get = i2s_mcux_config_get,
+	.read = i2s_mcux_read,
+	.write = i2s_mcux_write,
+	.trigger = i2s_mcux_trigger,
+};
+
+static void i2s_mcux_isr(const struct device *dev)
+{
+	const struct i2s_mcux_config *cfg = dev->config;
+	struct i2s_mcux_data *dev_data = dev->data;
+	struct stream *strm = &dev_data->dma_tx;
+	I2S_Type *base = cfg->base;
+	uint32_t intstat = base->FIFOINTSTAT;
+
+	if ((intstat & I2S_FIFOINTSTAT_TXERR_MASK) != 0UL) {
+		/* Clear TX error interrupt flag */
+		base->FIFOSTAT = I2S_FIFOSTAT_TXERR(1U);
+		strm = &dev_data->dma_tx;
+		strm->state = I2S_STATE_ERROR;
+	}
+
+	if ((intstat & I2S_FIFOINTSTAT_RXERR_MASK) != 0UL) {
+		/* Clear RX error interrupt flag */
+		base->FIFOSTAT = I2S_FIFOSTAT_RXERR(1U);
+		strm = &dev_data->dma_rx;
+		strm->state = I2S_STATE_ERROR;
+	}
+}
+
+static int i2s_mcux_init(const struct device *dev)
+{
+	const struct i2s_mcux_config *cfg = dev->config;
+	struct i2s_mcux_data *const data = dev->data;
+
+	cfg->irq_config(dev);
+
+	/* Initialize the buffer queues */
+	k_msgq_init(&data->dma_tx.in_queue, (char *)data->dma_tx.in_msgs,
+		    sizeof(void *), CONFIG_I2S_MCUX_FLEXCOMM_TX_BLOCK_COUNT);
+	k_msgq_init(&data->dma_rx.in_queue, (char *)data->dma_rx.in_msgs,
+		    sizeof(void *), CONFIG_I2S_MCUX_FLEXCOMM_RX_BLOCK_COUNT);
+	k_msgq_init(&data->dma_tx.out_queue, (char *)data->dma_tx.out_msgs,
+		    sizeof(void *), CONFIG_I2S_MCUX_FLEXCOMM_TX_BLOCK_COUNT);
+	k_msgq_init(&data->dma_rx.out_queue, (char *)data->dma_rx.out_msgs,
+		    sizeof(void *), CONFIG_I2S_MCUX_FLEXCOMM_RX_BLOCK_COUNT);
+
+	if (data->dma_tx.dev_dma != NULL) {
+		if (!device_is_ready(data->dma_tx.dev_dma)) {
+			LOG_ERR("%s device not ready", data->dma_tx.dev_dma->name);
+			return -ENODEV;
+		}
+	}
+
+	if (data->dma_rx.dev_dma != NULL) {
+		if (!device_is_ready(data->dma_rx.dev_dma)) {
+			LOG_ERR("%s device not ready", data->dma_rx.dev_dma->name);
+			return -ENODEV;
+		}
+	}
+
+	data->dma_tx.state = I2S_STATE_NOT_READY;
+	data->dma_rx.state = I2S_STATE_NOT_READY;
+
+	LOG_DBG("%s inited", dev->name);
```

Comment:
Could be changed to
```
	LOG_INF("Device %s initialized", dev->name);
```
It's actually useful for the user to get an indication that the device was initialized.

Reply:
Sure, I will make this change.